### DISCUSSION TITLE
Is it possible to know if a widget is visible in the viewport

### DISCUSSION THREAD

**Discussion by TingTingin** [flet-dev/flet#1833] [General]
When using a listview is it possible to know when a widget is visible in the viewport since widgets are loaded dynamically and when loading nothing is shown on the screen (especially when loading images) it would be helpful if you could query the viewport to know if the widgets are visible yet and and show a loading or waiting message etc

 I was able to find a widget on pub.dev https://pub.dev/packages/visibility detector but don't know if this is something you'd want to add

**Comment by ndonkoHenri** [comment]
For images, I suggest this: https://docs.flutter.dev/cookbook/images/fading-in-images.

**Comment by TingTingin** [comment]
That seems nice as well though I wanted more control in general it seems that if that feature was added it would only allow for fading in also I wonder how this works with a ListView in a app running locally as that's the use case I have right now when scrolling fast listview takes a while to render the images dont know if that can fix the problem since all the images are local files
